src/pages/Main/components/TokensBlock/TokenCard.tsx

```diff
@@ -32,64 +35,102 @@ const TokenCard: FC<TokenCardProps> = ({
 
   const imageUrl = getImageURL(imagePath);
 
+  const isImgUrlValid = validateUrl(imageUrl);
+
   return (
     <TokenCardLink
       onClick={onNFTCardClick}
       to={`/${currentChain.network}/tokens/${collectionId}/${tokenId}`}
     >
-      <TokenPicture
+      {!isImgUrlValid && <TokenPicture
         alt={tokenId.toString()}
         src={imageUrl}
-      />
+      />}
+
+      {isImgUrlValid && <TokenBackground imgUrl={imageUrl} />}
```

Comment:
Please use useImageLoader to check if the tokenImage exists on the server

Reply:
done